README
Introduction 

    PerlMagick, version 1.31, is an objected-oriented Perl interface to
    ImageMagick.  Use the module to read, manipulate, or write an image
    or image sequence from within a Perl script. This makes it suitable
    for Web CGI scripts. You must have ImageMagick 4.0 above and Perl
    version 5.002 or greater installed on your system.  Perl version
    5.004_04 or greater is required for PerlMagick to work under NT.

    See

      http://www.wizards.dupont.com/cristy/www/perl.html

    for additional information about PerlMagick.  See

      http://www.wizards.dupont.com/cristy/

    for instructions about installing ImageMagick.


Installation 

    Get the PerlMagick distribution and type the following: 

        gunzip PerlMagick-1.31.tar.gz
        tar xvf PerlMagick-1.31.tar
        cd Magick

    Next, edit Makefile.PL and change LIBS and INC to include the
    appropriate path information to the required libMagick library. You
    will also need paths to JPEG, PNG, TIFF, etc. libraries if they
    were included with your installed version of ImageMagick. Now type 

        perl Makefile.PL
        make
        make install

    Use nmake instead of make on an NT system.  For NT, you also need
    to copy IMagick.dll and X11.dll from the NT ImageMagick (see
    ftp://ftp.wizards.dupont.com/pub/ImageMagick/nt) distribution to a
    path library such as c:\perl\lib.

    For Unix, you typically need to be root to install the software.
    There are ways around this.  Consult the Perl manual pages for more
    information. You are now ready to utilize the PerlMagick routines
    from within your Perl scripts.


Example Perl Magick Script 

    Here is an example script to get you started: 

        #!/usr/local/bin/perl
        use Image::Magick;

        $q = Image::Magick->new;
        $x = $q->Read("girl.gif", "logo.gif", "rose.gif");
        warn "$x" if $x;

        $x = $q->Crop(geom=>'100x100+100+100');
        warn "$x" if $x;

        $x = $q->Write("x.gif");
        warn "$x" if $x;

    The script reads three images, crops them, and writes a single
    image as a GIF animation sequence.



Changelog
2013-12-01  6.87 Cristy  <quetzlzacatenango@image...>
  * Update PerlMagick to work with ImageMagick versions 6.8.7 and above.

2011-12-25  6.7.4-2 Cristy  <quetzlzacatenango@image...>
  * Relax when PerlMagick tosses an image when an exception is thrown.

2011-12-20  6.7.4-1 Cristy  <quetzlzacatenango@image...>
  * Use PTR2IV() and INT2PTR() macros to avoid int to ptr cast messages.

2011-08-25  6.7.2-0 Cristy  <quetzlzacatenango@image...>
  * Fix fault for Thumbnail() with GIF animation (problem report and patch
    from Shigeki Morimoto).

2011-08-09  6.7.1-4 Cristy  <quetzlzacatenango@image...>
  * Add a  simple "ignore this type" style typemap to get past a xsubpp
    exception in recent versions of Perl.

2011-06-06  6.7.0-5 Cristy  <quetzlzacatenango@image...>
  * Set modulate() colorspace predicated on parameter (e.g. brightness is HSB).

2011-01-25  6.6.7-3 Cristy  <quetzlzacatenango@image...>
  * Verify color parameter is a Perl array to avoid seg fault (reference
    http://www.imagemagick.org/discourse-server/viewtopic.php?f=7&t=17928).

2010-11-11  6.6.5-6 Cristy  <quetzlzacatenango@image...>
  * Check that image exists before accessing its member (reference
    http://www.imagemagick.org/discourse-server/viewtopic.php?f=2&t=17502).

2010-11-01  6.6.5-6 Cristy  <quetzlzacatenango@image...>
  * The gravity attribute must set the ImageInfo gravity property (reference
    http://www.imagemagick.org/discourse-server/viewtopic.php?f=3&t=17282).

2009-12-08  6.5.8-5 <kmx@volny...>
  * A couple of minor changes to Makefile.PL to support Strawberry Perl.

2009-12-06  6.5.8-4 Cristy  <quetzlzacatenango@image...>
  * Support the crop() gravity option.

2009-11-23  6.5.8-0 Cristy  <quetzlzacatenango@image...>
  * Add 'dither' option to Remap().

2009-10-18  6.5.7-0 Cristy  <quetzlzacatenango@image...>
  * Interpret gravity properly for ExtentImage().
  * Support the  Modulate() method parameter, 'lightness.'

2009-10-08  6.5.6-10 <kmx@volny...>
  * Improve PerlMagick.PL to work with Strawberry Perl under Windows.

2009-08-02  6.5.4-8 Cristy  <quetzlzacatenango@image...>
  * Create a image registry to track which image objects must be cloned
    when a Perl thread is created.

2009-06-14  6.5.3-7 Anthony Thyssen <A.Thyssen@griffith...>
  * Expand PerlMagick Composite() 'mask' merge for both compose 'displace'
    and new 'distort' mapping (absolute lookup)

2009-06-11  6.5.3-5 Anthony Thyssen <A.Thyssen@griffith...>
  * Fixed PerlMagick Composite() for correct merging of 'mask' image into
    the overlay image for Displacement Mapping Composition.

2009-06-02  6.5.3-2 Cristy  <quetzlzacatenango@image...>
  * Fix filename parameter in Read() (e.g. $image->Read(filename=>'xc:white')).  
2009-04-14  6.5.1-3 Cristy  <quetzlzacatenango@image...>
  * Support magick parameter for Read() (e.g.
    $im->Read( file => $fh, magick +> 'text:' )).

2008-11-16  6.4.4-5 Cristy  <quetzlzacatenango@image...>
  * Statistics() returns normalized values.

2008-10-16  6.4.4-9 Michael.Kroell <michael.kroell@gmail...>
  * Check if Image::Magick::constant is defined in AUTOLOAD() function.

2008-06-07  6.4.1-9 Cristy  <quetzlzacatenango@image...>
  * Permit settings of the form:
    $image->Set("quantum:polarity","min-is-white").

2008-04-07  6.4.0-5 Cristy  <quetzlzacatenango@image...>
  * X11 regression test passed if the DISPLAY variable is not defined (reference
		http://www.imagemagick.org/discourse-server/viewtopic.php?f=3&t=10979).
  * Make regression test, x11/write.t, exit after 8 seconds as stated (reference
		http://www.imagemagick.org/discourse-server/viewtopic.php?f=3&t=10979).

2007-11-07  6.3.7-0 Cristy  <quetzlzacatenango@image...>
  * Added gravity attribute to Transform() method.

2007-09-12  6.3.5-9 Cristy  <quetzlzacatenango@image...>
  * Added GetPixel()/SetPixel() methods.

2007-06-22  6.3.4-2 grafman <grafman@graphcomp.com>
  * Add GetImagePixels to return C pointer for image cache.
  * Add SyncImagePixels to sync image cache.
  * Add Quantum parameter to Get API to return Quantum-Depth.

2007-06-21  6.3.4-1 Cristy  <quetzlzacatenango@image...>
  * Add Distort.

2007-03-20  6.3.3-3 Cristy  <quetzlzacatenango@image...>
  * Add vector-graphics parameter to the Draw() method.

    blob length was not being set properly).
2007-03-11  6.3.3-2 Cristy  <quetzlzacatenango@image...>
  * Read(blob=>$blob) and Ping(blob=>$blob) sometimes failed because the
    blob length was not being set properly).

2007-01-23  cristy  <cristy@virgi...>

  * Add parameters fill-pattern and stroke-pattern to Annotate() and Draw().

2005-01-10  cristy  <cristy@virgi...>
  * Added normalize parameter to GetPixels().  Use normalize=>'true' to return
    normalized pixel values.


check.sh
#  Copyright 1999-2010 ImageMagick Studio LLC, a non-profit organization
#  dedicated to making software imaging solutions freely available.
#
#  You may not use this file except in compliance with the License.  You may
#  obtain a copy of the License at
#
#    http://www.imagemagick.org/script/license.php
#
#  Unless required by applicable law or agreed to in writing, software
#  distributed under the License is distributed on an "AS IS" BASIS,
#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#  See the License for the specific language governing permissions and
#  limitations under the License.
# 
#  Copyright (C) 2003 - 2009 GraphicsMagick Group
#
#  Check script for building PerlMagick.

echo "LD_LIBRARY_PATH='${LD_LIBRARY_PATH}'"
echo "MAGICK_CODER_MODULE_PATH='${MAGICK_CODER_MODULE_PATH}'"
echo "MAGICK_CONFIGURE_PATH='${MAGICK_CONFIGURE_PATH}'"
echo "MAGICK_FILTER_MODULE_PATH='${MAGICK_FILTER_MODULE_PATH}'"
echo "MAKE='${MAKE}'"
echo "MAKEFLAGS='${MAKEFLAGS}'"
echo "MEMCHECK='${MEMCHECK}'"
echo "PATH='${PATH}'"
echo "SRCDIR='${SRCDIR}'"
echo "srcdir='${srcdir}'"

set -x

SRCDIR=`dirname $0`
SRCDIR=`cd $SRCDIR && pwd`
TOPSRCDIR=`cd $srcdir && pwd`

cd PerlMagick || exit 1

if test -z "${MAKE}" ; then
  MAKE=make
fi

if test -x PerlMagick -a -f Makefile.aperl ; then
  # Static build test incantation
  ${MAKE} -f Makefile.aperl CC='gcc -std=gnu99' test
elif test -f Makefile -a -f Magick.o; then
  # Shared build test incantation
  ${MAKE} CC='gcc -std=gnu99' test
else
  echo 'PerlMagick has not been built!'
  exit 1
fi


Magick.pm
package Image::Magick;

#  Copyright 1999-2013 ImageMagick Studio LLC, a non-profit organization
#  dedicated to making software imaging solutions freely available.
#
#  You may not use this file except in compliance with the License.  You may
#  obtain a copy of the License at
#
#    http://www.imagemagick.org/script/license.php
#
#  Unless required by applicable law or agreed to in writing, software
#  distributed under the License is distributed on an "AS IS" BASIS,
#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#  See the License for the specific language governing permissions and
#  limitations under the License.
#
#  Initial version, written by Kyle Shorter.

use strict;
use Carp;
use vars qw($VERSION @ISA @EXPORT $AUTOLOAD);

require 5.002;
require Exporter;
require DynaLoader;
require AutoLoader;

@ISA = qw(Exporter DynaLoader);
# Items to export into callers namespace by default. Note: do not export
# names by default without a very good reason. Use EXPORT_OK instead.
# Do not simply export all your public functions/methods/constants.
@EXPORT =
  qw(
      Success Transparent Opaque QuantumDepth QuantumRange MaxRGB
      WarningException ResourceLimitWarning TypeWarning OptionWarning
      DelegateWarning MissingDelegateWarning CorruptImageWarning
      FileOpenWarning BlobWarning StreamWarning CacheWarning CoderWarning
      ModuleWarning DrawWarning ImageWarning XServerWarning RegistryWarning
      ConfigureWarning ErrorException ResourceLimitError TypeError
      OptionError DelegateError MissingDelegateError CorruptImageError
      FileOpenError BlobError StreamError CacheError CoderError
      ModuleError DrawError ImageError XServerError RegistryError
      ConfigureError FatalErrorException
    );

$VERSION = '6.87';

sub AUTOLOAD {
    # This AUTOLOAD is used to 'autoload' constants from the constant()
    # XS function.  If a constant is not found then control is passed
    # to the AUTOLOAD in AutoLoader.

    my $constname;
    ($constname = $AUTOLOAD) =~ s/.*:://;
    die "&${AUTOLOAD} not defined. The required ImageMagick libraries are not installed or not installed properly.\n" if $constname eq 'constant';
    my $val = constant($constname, @_ ? $_[0] : 0);
    if ($! != 0) {
    	if ($! =~ /Invalid/) {
	        $AutoLoader::AUTOLOAD = $AUTOLOAD;
	        goto &AutoLoader::AUTOLOAD;
    	}
    	else {
	        my($pack,$file,$line) = caller;
	        die "Your vendor has not defined PerlMagick macro $pack\:\:$constname, used at $file line $line.\n";
    	}
    }
    eval "sub $AUTOLOAD { $val }";
    goto &$AUTOLOAD;
}

bootstrap Image::Magick $VERSION;

# Preloaded methods go here.

sub new
{
    my $this = shift;
    my $class = ref($this) || $this || "Image::Magick";
    my $self = [ ];
    bless $self, $class;
    $self->set(@_) if @_;
    return $self;
}

sub New
{
    my $this = shift;
    my $class = ref($this) || $this || "Image::Magick";
    my $self = [ ];
    bless $self, $class;
    $self->set(@_) if @_;
    return $self;
}

# Autoload methods go after =cut, and are processed by the autosplit program.

END { UNLOAD () };

1;
__END__

=head1 NAME

Image::Magick - objected-oriented Perl interface to ImageMagick. Use it to read, manipulate, or write an image or image sequence from within a Perl script.

=head1 SYNOPSIS

  use Image::Magick;
  $p = new Image::Magick;
  $p->Read("imagefile");
  $p->Set(attribute => value, ...)
  ($a, ...) = $p->Get("attribute", ...)
  $p->routine(parameter => value, ...)
  $p->Mogrify("Routine", parameter => value, ...)
  $p->Write("filename");

=head1 DESCRIPTION

This Perl extension allows the reading, manipulation and writing of
a large number of image file formats using the ImageMagick library.
It was originally developed to be used by CGI scripts for Web pages.

A web page has been set up for this extension. See:

	 file:///usr/local/share/doc/ImageMagick-6.8.2/www/perl-magick.html
	 http://www.imagemagick.org/script/perl-magick.php

If you have problems, go to

   http://www.imagemagick.org/discourse-server/viewforum.php?f=7

=head1 AUTHOR

Kyle Shorter	magick-users@imagemagick.org

=head1 BUGS

Has all the bugs of ImageMagick and much, much more!

=head1 SEE ALSO

perl(1).

=cut


magick.sh
# !/bin/sh
#
#  Copyright 1999-2006 ImageMagick Studio LLC, a non-profit organization
#  dedicated to making software imaging solutions freely available.
#
#  You may not use this file except in compliance with the License.  You may
#  obtain a copy of the License at
#
#    http://www.imagemagick.org/script/license.php
#
#  Unless required by applicable law or agreed to in writing, software
#  distributed under the License is distributed on an "AS IS" BASIS,
#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#  See the License for the specific language governing permissions and
#  limitations under the License.
#
#  Top-Level Makefile for building ImageMagick.

top_srcdir='/home/cristy/ImageMagick-6.4.0'
top_builddir='/home/cristy/ImageMagick-6.4.0'

MAGICK_CODER_MODULE_PATH='/home/cristy/ImageMagick-6.4.0/coders'
MAGICK_CONFIGURE_SRC_PATH='/home/cristy/ImageMagick-6.4.0/config'
MAGICK_CONFIGURE_BUILD_PATH='/home/cristy/ImageMagick-6.4.0/config'
MAGICK_FILTER_MODULE_PATH='/home/cristy/ImageMagick-6.4.0/filters'
DIRSEP=':'

PATH="${top_builddir}/utilities:${PATH}"

if test -n "$VERBOSE"
then
  echo "$@"
fi
env \
  LD_LIBRARY_PATH="${top_builddir}/magick/.libs:${top_builddir}/wand/.libs:${LD_LIBRARY_PATH}" \
  MAGICK_CODER_MODULE_PATH="${MAGICK_CODER_MODULE_PATH}" \
  MAGICK_CONFIGURE_PATH="${MAGICK_CONFIGURE_BUILD_PATH}${DIRSEP}${MAGICK_CONFIGURE_SRC_PATH}" \
  MAGICK_FILTER_MODULE_PATH="${MAGICK_FILTER_MODULE_PATH}" \
  PATH="${PATH}" \
  "$@"


demo/demo.pl
#!/usr/bin/perl
#
# Overall demo of the major PerlMagick methods.
#
use Image::Magick;

#
# Read model & smile image.
#
print "Read...\n";
$null=Image::Magick->new;
$null->Set(size=>'70x70');
$x=$null->ReadImage('NULL:black');
warn "$x" if "$x";

$model=Image::Magick->new();
$x=$model->ReadImage('model.gif');
warn "$x" if "$x";
$model->Label('Magick');
$model->Set(background=>'white');

$smile=Image::Magick->new;
$x=$smile->ReadImage('smile.gif');
warn "$x" if "$x";
$smile->Label('Smile');
$smile->Set(background=>'white');
#
# Create image stack.
#
print "Transform image...\n";
$images=Image::Magick->new();

print "Adaptive Blur...\n";
$example=$model->Clone();
$example->Label('Adaptive Blur');
$example->AdaptiveBlur('0x1');
push(@$images,$example);

print "Adaptive Resize...\n";
$example=$model->Clone();
$example->Label('Adaptive Resize');
$example->AdaptiveResize('60%');
push(@$images,$example);

print "Adaptive Sharpen...\n";
$example=$model->Clone();
$example->Label('Adaptive Sharpen');
$example->AdaptiveSharpen('0x1');
push(@$images,$example);

print "Adaptive Threshold...\n";
$example=$model->Clone();
$example->Label('Adaptive Threshold');
$example->AdaptiveThreshold('5x5+5%');
push(@$images,$example);

print "Add Noise...\n";
$example=$model->Clone();
$example->Label('Add Noise');
$example->AddNoise("Laplacian");
push(@$images,$example);

print "Annotate...\n";
$example=$model->Clone();
$example->Label('Annotate');
$example->Annotate(text=>'Magick',geometry=>'+0+20',font=>'Generic.ttf',
  fill=>'gold',gravity=>'North',pointsize=>14);
push(@$images,$example);

print "Blur...\n";
$example=$model->Clone();
$example->Label('Blur');
$example->Blur('0.0x1.0');
push(@$images,$example);

print "Border...\n";
$example=$model->Clone();
$example->Label('Border');
$example->Border(geometry=>'6x6',color=>'gold');
push(@$images,$example);

print "Channel...\n";
$example=$model->Clone();
$example->Label('Channel');
$example->Channel(channel=>'red');
push(@$images,$example);

print "Charcoal...\n";
$example=$model->Clone();
$example->Label('Charcoal');
$example->Charcoal('0x1');
push(@$images,$example);

print "Composite...\n";
$example=$model->Clone();
$example->Label('Composite');
$example->Composite(image=>$smile,compose=>'over',geometry=>'+35+65');
push(@$images,$example);

print "Contrast...\n";
$example=$model->Clone();
$example->Label('Contrast');
$example->Contrast();
push(@$images,$example);

print "Contrast Stretch...\n";
$example=$model->Clone();
$example->Label('Contrast Stretch');
$example->ContrastStretch('5%');

push(@$images,$example);
print "Convolve...\n";
$example=$model->Clone();
$example->Label('Convolve');
$example->Convolve([1, 1, 1, 1, 4, 1, 1, 1, 1]);
push(@$images,$example);

print "Crop...\n";
$example=$model->Clone();
$example->Label('Crop');
$example->Crop(geometry=>'80x80+25+50');
$example->Set(page=>'0x0+0+0');
push(@$images,$example);

print "Despeckle...\n";
$example=$model->Clone();
$example->Label('Despeckle');
$example->Despeckle();
push(@$images,$example);

print "Draw...\n";
$example=$model->Clone();
$example->Label('Draw');
$example->Draw(fill=>'none',stroke=>'gold',primitive=>'circle',
  points=>'60,90 60,120',strokewidth=>2);
push(@$images,$example);

print "Detect Edges...\n";
$example=$model->Clone();
$example->Label('Detect Edges');
$example->Edge();
push(@$images,$example);

print "Emboss...\n";
$example=$model->Clone();
$example->Label('Emboss');
$example->Emboss('0x1');
push(@$images,$example);

print "Equalize...\n";
$example=$model->Clone();
$example->Label('Equalize');
$example->Equalize();
push(@$images,$example);

print "Implode...\n";
$example=$model->Clone();
$example->Label('Explode');
$example->Implode(-1);
push(@$images,$example);

print "Flip...\n";
$example=$model->Clone();
$example->Label('Flip');
$example->Flip();
push(@$images,$example);

print "Flop...\n";
$example=$model->Clone();
$example->Label('Flop');
$example->Flop();
push(@$images,$example);

print "Frame...\n";
$example=$model->Clone();
$example->Label('Frame');
$example->Frame('15x15+3+3');
push(@$images,$example);

print "Fx...\n";
$example=$model->Clone();
$example->Label('Fx');
push(@$images,$example->Fx(expression=>'0.5*u'));

print "Gamma...\n";
$example=$model->Clone();
$example->Label('Gamma');
$example->Gamma(1.6);
push(@$images,$example);

print "Gaussian Blur...\n";
$example=$model->Clone();
$example->Label('Gaussian Blur');
$example->GaussianBlur('0.0x1.5');
push(@$images,$example);

print "Gradient...\n";
$gradient=Image::Magick->new;
$gradient->Set(size=>'130x194');
$x=$gradient->ReadImage('gradient:#20a0ff-#ffff00');
warn "$x" if "$x";
$gradient->Label('Gradient');
push(@$images,$gradient);

print "Grayscale...\n";
$example=$model->Clone();
$example->Label('Grayscale');
$example->Quantize(colorspace=>'gray');
push(@$images,$example);

print "Implode...\n";
$example=$model->Clone();
$example->Label('Implode');
$example->Implode(0.5);
push(@$images,$example);

print "Level...\n";
$example=$model->Clone();
$example->Label('Level');
$example->Level(level=>'20%');
push(@$images,$example);

print "Median Filter...\n";
$example=$model->Clone();
$example->Label('Median Filter');
$example->MedianFilter();
push(@$images,$example);

print "Modulate...\n";
$example=$model->Clone();
$example->Label('Modulate');
$example->Modulate(brightness=>110,saturation=>110,hue=>110);
push(@$images,$example);
$example=$model->Clone();

print "Monochrome...\n";
$example->Label('Monochrome');
$example->Quantize(colorspace=>'gray',colors=>2,dither=>'false');
push(@$images,$example);

print "Motion Blur...\n";
$example=$model->Clone();
$example->Label('Motion Blur');
$example->MotionBlur('0x13+10-10');
push(@$images,$example);

print "Negate...\n";
$example=$model->Clone();
$example->Label('Negate');
$example->Negate();
push(@$images,$example);

print "Normalize...\n";
$example=$model->Clone();
$example->Label('Normalize');
$example->Normalize();
push(@$images,$example);

print "Oil Paint...\n";
$example=$model->Clone();
$example->Label('Oil Paint');
$example->OilPaint();
push(@$images,$example);

print "Plasma...\n";
$plasma=Image::Magick->new;
$plasma->Set(size=>'130x194');
$x=$plasma->ReadImage('plasma:fractal');
warn "$x" if "$x";
$plasma->Label('Plasma');
push(@$images,$plasma);

print "Quantize...\n";
$example=$model->Clone();
$example->Label('Quantize');
$example->Quantize();
push(@$images,$example);

print "Radial Blur...\n";
$example=$model->Clone();
$example->Label('Radial Blur');
$example->RadialBlur(10);
push(@$images,$example);

print "Raise...\n";
$example=$model->Clone();
$example->Label('Raise');
$example->Raise('10x10');
push(@$images,$example);

print "Reduce Noise...\n";
$example=$model->Clone();
$example->Label('Reduce Noise');
$example->ReduceNoise();
push(@$images,$example);

print "Resize...\n";
$example=$model->Clone();
$example->Label('Resize');
$example->Resize('60%');
push(@$images,$example);

print "Roll...\n";
$example=$model->Clone();
$example->Label('Roll');
$example->Roll(geometry=>'+20+10');
push(@$images,$example);

print "Rotate...\n";
$example=$model->Clone();
$example->Label('Rotate');
$example->Rotate(45);
push(@$images,$example);

print "Sample...\n";
$example=$model->Clone();
$example->Label('Sample');
$example->Sample('60%');
push(@$images,$example);

print "Scale...\n";
$example=$model->Clone();
$example->Label('Scale');
$example->Scale('60%');
push(@$images,$example);

print "Segment...\n";
$example=$model->Clone();
$example->Label('Segment');
$example->Segment();
push(@$images,$example);

print "Shade...\n";
$example=$model->Clone();
$example->Label('Shade');
$example->Shade(geometry=>'30x30',gray=>'true');
push(@$images,$example);

print "Sharpen...\n";
$example=$model->Clone();
$example->Label('Sharpen');
$example->Sharpen('0.0x1.0');
push(@$images,$example);

print "Shave...\n";
$example=$model->Clone();
$example->Label('Shave');
$example->Shave('10x10');
push(@$images,$example);

print "Shear...\n";
$example=$model->Clone();
$example->Label('Shear');
$example->Shear('-20x20');
push(@$images,$example);

print "Sigmoidal Contrast...\n";
$example=$model->Clone();
$example->Label('Sigmoidal Contrast');
$example->SigmoidalContrast("3x50%");
push(@$images,$example);

print "Spread...\n";
$example=$model->Clone();
$example->Label('Spread');
$example->Spread();
push(@$images,$example);

print "Solarize...\n";
$example=$model->Clone();
$example->Label('Solarize');
$example->Solarize();
push(@$images,$example);

print "Swirl...\n";
$example=$model->Clone();
$example->Label('Swirl');
$example->Swirl(90);
push(@$images,$example);

print "Unsharp Mask...\n";
$example=$model->Clone();
$example->Label('Unsharp Mask');
$example->UnsharpMask('0.0x1.0');
push(@$images,$example);

print "Vignette...\n";
$example=$model->Clone();
$example->Label('Vignette');
$example->Vignette('0x20');
push(@$images,$example);

print "Wave...\n";
$example=$model->Clone();
$example->Label('Wave');
$example->Wave('25x150');
push(@$images,$example);
#
# Create image montage.
#
print "Montage...\n";
$montage=$images->Montage(geometry=>'128x160+8+4>',gravity=>'Center',
  tile=>'5x+10+200',compose=>'over',background=>'#ffffff',
  font=>'Generic.ttf',pointsize=>18,fill=>'#600',stroke=>'none');

$logo=Image::Magick->new();
$logo->Read('logo:');
$logo->Zoom('40%');
$montage->Composite(image=>$logo,gravity=>'North');

print "Write...\n";
$montage->Set(matte=>'false');
$montage->Write('demo.jpg');
print "Display...\n";
$montage->Write('win:');
